src/graphql/dataLoaders/urlLoaderFactory.js

```diff
@@ -0,0 +1,32 @@
+import DataLoader from 'dataloader';
+
+/**
+ * Given list of urls, return their latest fetch respectively
+ */
+export default dataLoaders =>
+  new DataLoader(
+    async urls => {
+      const data = await dataLoaders.searchResultLoader.loadMany(
+        urls.map(url => ({
+          index: 'urls',
+          type: 'doc',
+          body: {
+            query: {
+              bool: {
+                should: [{ term: { url } }, { term: { canonical: url } }],
+              },
+            },
+            sort: {
+              fetchedAt: 'desc',
+            },
+            size: 1,
+          },
+        }))
+      );
+
+      return data.map(result => (result && result.length ? result[0] : null));
+    },
+    {
+      cacheKeyFn: JSON.stringify,
+    }
```

Comment:
since `url` is already serializable by itself, we don't need `cacheKeyFn` actually. It will be handled in [a future PR](https://github.com/cofacts/rumors-api/pull/97/files#diff-2e55e46fc2778c618445a5f8c6943066L32).